CMakeLists.txt

```diff
@@ -161,8 +161,12 @@ endif()
 if(LINK_WITH_PYTHON)
   # find_package(Python) has the problem that on github actions in particular it'll pick up the most recent python (eg 3.9) from the tool cache
   # even if you have used the setup-python action and set it to 3.8, so we make the exact version required
-  if(PYTHON_VERSION)
-    find_package(Python ${PYTHON_VERSION} EXACT COMPONENTS Interpreter Development REQUIRED)
+  set(Python_REQUIRED_VERSION "" CACHE STRING "The specific Python version you want to link to (Optional)")
+  set(Python_ROOT_DIR "" CACHE PATH "The path to the root installation of your python version, if you want to explicitly set it (Optional)")
+  mark_as_advanced(Python_REQUIRED_VERSION)
+  mark_as_advanced(Python_ROOT_DIR)
+  if(Python_REQUIRED_VERSION)
+    find_package(Python ${Python_REQUIRED_VERSION} EXACT COMPONENTS Interpreter Development REQUIRED)
```

Comment:
Do we still need to expose `PYTHON_INCLUDE_DIR`, `PYTHON_LIBRARY`, and `PYTHON_LIBRARY_DEBUG` or are those no longer needed?

Reply:
@mitchute Those are artifacts from when the deprecated `find_package(PythonInterp)` and `find_package(PythonLibs)` were used. You don't see them on a clean build. 

cf https://cmake.org/cmake/help/latest/module/FindPythonLibs.html